logstash-core/src/main/java/org/logstash/settings/SettingPortRange.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.logstash.settings;
+
+import org.jruby.RubyInteger;
+import org.jruby.RubyRange;
+import org.logstash.RubyUtil;
+
+// Ideally would be a Coercible<Range<Integer>>, but given the fact that
+// values can be effectively coerced into the constructor, it needs instances
+// of Objects to represent Integer, String, Long to be later coerced into Range<Integer>.
+@SuppressWarnings({"rawtypes", "unchecked"})
+public class SettingPortRange extends Coercible<Object> {
```

Comment:
Could we name these classes `PortRangeSetting` instead of `SettingPortRange`? Typically Java class naming goes from more specific to more generic: RuntimeException, ConcurrentHashMap, ThreadPoolExecutor, etc.

Reply:
You are right, was named like this to be consistent with all the existing classes already moved https://github.com/elastic/logstash/issues/17157.

Wa can rename it and create an issue to adjust the existing ones. WDYT?